Submitted By:     VMiklos <vmiklos@frugalware.org>
Date:             2005.01.21
Initial Packages: 2.9.4
Upstream Status:  Not submitted - Frugalware specific
Description:      This patch adds a new feature to makepkg.
                  Automatically update the last modified line in FrugalBuild
                  after a successfully build.

diff -Naur pacman-2.9.6.orig/doc/makepkg.8.in pacman-2.9.6/doc/makepkg.8.in
--- pacman-2.9.6.orig/doc/makepkg.8.in	2004-08-04 07:57:17.000000000 +0200
+++ pacman-2.9.6/doc/makepkg.8.in	2005-08-02 21:31:54.000000000 +0200
@@ -362,6 +362,9 @@
 Sets MAKEFLAGS="-j<jobs>" before building the package.  This is useful for overriding
 the MAKEFLAGS setting in /etc/makepkg.conf.
 .TP
+.B "\-L, \-\-nolastmod"
+Disable updating the Last modified line.
+.TP
 .B "\-m, \-\-nocolor"
 Disable color in output messages
 .TP
diff -Naur pacman-2.9.6.orig/scripts/makepkg pacman-2.9.6/scripts/makepkg
--- pacman-2.9.6.orig/scripts/makepkg	2005-06-10 22:54:29.000000000 +0200
+++ pacman-2.9.6/scripts/makepkg	2005-08-02 21:30:17.000000000 +0200
@@ -185,6 +185,7 @@
 	echo "  -h, --help       This help"
 	echo "  -i, --install    Install package after successful build"
 	echo "  -j <jobs>        Set MAKEFLAGS to \"-j<jobs>\" before building"
+	echo "  -L, --nolastmod  Do not update the \"Last modified\" line"
 	echo "  -m, --nocolor    Disable colorized output messages"
 	echo "  -n, --nostrip    Do not strip binaries/libraries"
 	echo "  -o, --nobuild    Download and extract files only"
@@ -203,6 +204,7 @@
 FORCE=0
 GENMD5=0
 INSTALL=0
+NOLASTMOD=0
 NOBUILD=0
 NOCCACHE=0
 NODEPS=0
@@ -225,6 +227,7 @@
 		--nodeps)     NODEPS=1 ;;
 		--noextract)  NOEXTRACT=1 ;;
 		--install)    INSTALL=1 ;;
+		--nolastmod)  NOLASTMOD=1 ;;
 		--force)      FORCE=1 ;;
 		--nostrip)    NOSTRIP=1 ;;
 		--nobuild)    NOBUILD=1 ;;
@@ -240,7 +243,7 @@
 			exit 1
 			;;
 		-*)
-			while getopts "bBcCdefghij:mnop:rsSw:-" opt; do
+			while getopts "bBcCdefghij:Lmnop:rsSw:-" opt; do
 				case $opt in
 					c) CLEANUP=1 ;;
 					C) CLEANCACHE=1 ;;
@@ -255,6 +258,7 @@
 						;;
 					i) INSTALL=1 ;;
 					j) export MAKEFLAGS="-j$OPTARG" ;;
+					L) NOLASTMOD=1 ;;
 					m) USE_COLOR="n" ;;
 					n) NOSTRIP=1 ;;
 					o) NOBUILD=1 ;;
@@ -690,6 +694,11 @@
 	pacman -R $makedeplist $deplist
 fi
 
+# update last mod line
+if [ "$NOLASTMOD" = "0" ]; then
+	sed -i "s/^\(# Last modified: \).*/\1`date -R`/i" $BUILDSCRIPT
+fi
+
 msg "Finished making: $pkgname  (`date`)"
 
 if [ "$INSTALL" = "1" -a "`id -u`" = "0" -a "$INFAKEROOT" != "1" ]; then
